pyrefly/lib/test/lsp/completion.rs

```diff
@@ -149,6 +169,81 @@ Completion Results:
     );
 }
 
+#[test]
+fn dict_key_completion_from_literal() {
+    let code = r#"
+def use_dict():
+    data = {"foo": 1, "bar": 2}
+    data[""]
+#        ^
+"#;
+    let report =
+        get_batched_lsp_operations_report_allow_error(&[("main", code)], get_default_test_report());
+    let report = strip_ansi(&report);
+    assert!(
+        report.trim().contains(
+            r#"
+# main.py
+4 |     data[""]
+             ^
+Completion Results:
+- (Field) bar: Literal[2]
+- (Field) foo: Literal[1]
+"#
+            .trim()
+        )
+    );
+}
+
+#[test]
+fn dict_key_completion_from_nested_literal() {
+    let code = r#"
+def nested():
+    config = {"user": {"name": "Alice"}}
+    config["user"][""]
+#               ^
+"#;
+    let (handles, state) = mk_multi_file_state(&[("main", code)], Require::indexing(), true);
+    let handle = handles.get("main").unwrap();
+    let position = extract_cursors_for_test(code)[0];
+    let completions = state
+        .transaction()
+        .completion(handle, position, ImportFormat::Absolute);
+    let labels: Vec<_> = completions.iter().map(|item| item.label.as_str()).collect();
+    assert!(labels.contains(&"name"));
```

Comment:
does it seem weird that we're testing that the completions on `user` result in the `name` completion? is it that we get completions for every nested dict value on all of them and the language client is supposed to filter it